test/tls-provider.c

```diff
@@ -748,89 +1274,2261 @@ static const OSSL_DISPATCH xor_keymgmt_functions[] = {
+
+static int xor_get_aid(unsigned char** oidbuf, const char *tls_name) {
+   X509_ALGOR *algor = X509_ALGOR_new();
+   int aidlen = 0;
+
+   X509_ALGOR_set0(algor, OBJ_txt2obj(tls_name, 0), V_ASN1_UNDEF, NULL);
+
+   aidlen = i2d_X509_ALGOR(algor, oidbuf); 
+   X509_ALGOR_free(algor);
+   return(aidlen);
+}
+
+/*
+ * What's passed as an actual key is defined by the KEYMGMT interface.
+ */
+typedef struct {
+    OSSL_LIB_CTX *libctx;
+    char *propq;
+    XORKEY *sig;
+
+    /*
+     * Flag to determine if the hash function can be changed (1) or not (0)
+     * Because it's dangerous to change during a DigestSign or DigestVerify
+     * operation, this flag is cleared by their Init function, and set again
+     * by their Final function.
+     */
+    unsigned int flag_allow_md : 1;
+
+    char mdname[OSSL_MAX_NAME_SIZE];
+
+    /* The Algorithm Identifier of the combined signature algorithm */
+    unsigned char *aid;
+    size_t  aid_len;
+
+    /* main digest */
+    EVP_MD *md;
+    EVP_MD_CTX *mdctx;
+    int operation;
+} PROV_XORSIG_CTX;
+
+static void *xor_sig_newctx(void *provctx, const char *propq)
+{
+    PROV_XORSIG_CTX *pxor_sigctx;
+
+    pxor_sigctx = OPENSSL_zalloc(sizeof(PROV_XORSIG_CTX));
+    if (pxor_sigctx == NULL)
+        return NULL;
+
+    pxor_sigctx->libctx = ((PROV_XOR_CTX*)provctx)->libctx;
+    pxor_sigctx->flag_allow_md = 0;
+    if (propq != NULL && (pxor_sigctx->propq = OPENSSL_strdup(propq)) == NULL) {
+        OPENSSL_free(pxor_sigctx);
+        pxor_sigctx = NULL;
+        ERR_raise(ERR_LIB_USER, ERR_R_MALLOC_FAILURE);
+    }
+    return pxor_sigctx;
+}
+
+static int xor_sig_setup_md(PROV_XORSIG_CTX *ctx,
+                        const char *mdname, const char *mdprops)
+{
+    EVP_MD *md;
+
+    if (mdprops == NULL)
+        mdprops = ctx->propq;
+
+    md = EVP_MD_fetch(ctx->libctx, mdname, mdprops);
+
+    if ((md == NULL)||(EVP_MD_nid(md)==NID_undef)) {
+        if (md == NULL)
+            ERR_raise_data(ERR_LIB_USER, XORPROV_R_INVALID_DIGEST,
+                           "%s could not be fetched", mdname);
+        EVP_MD_free(md);
+        return 0;
+    }
+
+    EVP_MD_CTX_free(ctx->mdctx);
+    ctx->mdctx = NULL;
+    EVP_MD_free(ctx->md);
+    ctx->md = NULL;
+
+    OPENSSL_free(ctx->aid);
+    ctx->aid = NULL;
+    ctx->aid_len = xor_get_aid(&(ctx->aid), ctx->sig->tls_name);
+
+    ctx->mdctx = NULL;
+    ctx->md = md;
+    OPENSSL_strlcpy(ctx->mdname, mdname, sizeof(ctx->mdname));
+    return 1;
+}
+
+static int xor_sig_signverify_init(void *vpxor_sigctx, void *vxorsig,
+                                   int operation)
+{
+    PROV_XORSIG_CTX *pxor_sigctx = (PROV_XORSIG_CTX *)vpxor_sigctx;
+
+    if ( pxor_sigctx == NULL || vxorsig == NULL)
+        return 0;
+    xor_freekey(pxor_sigctx->sig);
+    if (!xor_key_up_ref(vxorsig)) return 0;
+    pxor_sigctx->sig = vxorsig;
+    pxor_sigctx->operation = operation;
+    if ( (operation==EVP_PKEY_OP_SIGN && !pxor_sigctx->sig) ||
```

Comment:
Space around `==` (here and below). Also make the second part `pxor_sigctx->sig == NULL`